src/apps/backend/modules/vendor_account/internal/vendor_account_writer.py

```diff
@@ -0,0 +1,33 @@
+from bson import ObjectId
+from datetime import datetime, timezone
+
+from modules.vendor_account.internal.store.vendor_account_repository import VendorAccountRepository
+from modules.vendor_account.internal.vendor_account_reader import VendorAccountReader
+from modules.vendor_account.internal.vendor_account_util import VendorAccountUtil
+from modules.vendor_account.types import CreateVendorAccountParams
+from modules.vendor_account.types import VendorAccount
+from modules.vendor_account.errors import VendorAccountWithSameNameAndAccountExistsError
+
+
+class VendorAccountWriter:
+    @staticmethod
+    def create_vendor_account(params: CreateVendorAccountParams) -> VendorAccount:
+        vendor_account = VendorAccountReader.get_vendor_account_optional(
+            account_id=params.account_id, vendor_account_name=params.name
```

Comment:
I believe you also want to scope the search by `vendor_type`. Maybe Ok to have same name across different vendors. You might have to update the index and add new tests around this as well